mike-petrut-2021-04-ndvi-automation.py

```diff
@@ -0,0 +1,743 @@
+
+
+# In[3]:
+
+
+# DO NOT MODIFY THIS CELL
+# Tests that the working directory is set to earth-analytics/data
+
+path = os.path.normpath(os.getcwd())
+student_wd_parts = path.split(os.sep)
+
+if student_wd_parts[-2:] == ['earth-analytics', 'data']:
+    print("\u2705 Great - it looks like your working directory is set correctly to ~/earth-analytics/data")
+else:
+    print("\u274C Oops, the autograder will not run unless your working directory is set to earth-analytics/data")
+
+
+# # Figure 1: Plot 1 - Mean NDVI For Each Site Across the Year (50 points)
+# 
+# Create a plot of the mean normalized difference vegetation index (NDVI) for the two different sites in the United States across the year: 
+# 
+# * NDVI on the x axis and formatted dates on the y for both NEON sites on one figure/axis object.
+# * Each site should be identified with a different color in the plot and legend.
+# * The final plot **data should be cleaned to remove the influence of clouds**.
+# * Be sure to include appropriate title and axes labels.
+# 
+# Add additional cells as needed for processing data (e.g. defining functions, etc), but be sure to:
+# * follow the instructions in the code cells that have been provided to ensure that you are able to use the sanity check tests that are provided. 
+# * include only the plot code in the cell identified for the final plot code below
+
+# ## Task 1: 
+# 
+# In the cell below, create a single dataframe containing MEAN NDVI, the site name, 
+# and the date of the data for the HARV site 
+# scene `HARV/landsat-crop/LC080130302017031701T1-SC20181023151837`.  The column names for the  final
+# DataFrame should be`mean_ndvi`, and `site`, and the data should be **indexed on the date**. 
+# 
+# Use the functions that we reviewed in class (or create your own versions of them) to implement your code
+# 
+# ### In the Cell below Place  All Functions Needed to Run this Notebook (20 points)
+
+# In[4]:
+
+
+### DO NOT REMOVE THIS LINE OR EDIT / MOVE THIS CELL ###
+start_time = datetime.now()
+
+
+# In[5]:
+
+
+# In this cell place all of the functions needed to run your notebook
+# You will be graded here on function application, docstrings, efficiency so ensure
+# All functions are placed here!
+
+def combine_tifs(tif_list):
+    
+    """A function that combines a list of tifs in the same CRS
+    and of the same extent into an xarray object
+
+    Parameters
+    ----------
+    tif_list : list
+        A list of paths to the tif files that you wish to combine.
+
+    Returns
+    -------
+    An xarray object with all of the tif files in the listmerged into 
+    a single object.
+
+    """
+    out_xr = []
+    for i, tif_path in enumerate(tif_list):
+        out_xr.append(rxr.open_rasterio(tif_path, masked = True).squeeze())
+        out_xr[i]["band"] = i + 1
+
+    return xr.concat(out_xr, dim = "band")
+
+def ndvi_mean(main_path, cloud_mask):
+    
+    """A function that takes a file path of Landsat imagery (which includes 
```

Comment:
Great function! Maybe it would be more reusable if it didn't do so many things though. Maybe calculating NDVI could be a separate function.

Reply:
@mred1208 agreed! Looking back not sure what I was thinking making it so long and overly complicated